docs/docs/articles/test-workflows-examples-basics.md

```diff
@@ -0,0 +1,417 @@
+# Test Workflows Examples - Basics
+
+## Running the Image
+
+The Test Workflows spec allows you to provide instructions in setup, steps and after properties.
+
+### Syntax
+
+Both setup, steps and after have the same syntax - they are a list of steps. The step is an object that has execution instructions.
+
+```yaml
+apiVersion: testworkflows.testkube.io/v1
+kind: TestWorkflow
+metadata:
+  name: overview--example-1
+spec:
+  steps:
+  - run:
+      image: 'curlimages/curl:7.78.0'
+      args:
+      - 'https://testkube.io'
+```
+### Running the Image
+
+Use a run instruction that has similar syntax to the native Kubernetes’ container. A command exiting with code > 0 is considered a failure.
+
+![Running the Image](../img/running-image.png)
+
+
+## Using Shell
+
+### Running Shell Commands
+
+For simplicity, you can also run shell commands with the **run** instruction and the **shell** property.
+The script is automatically prepended with **set -e**, so, by default, it will fail the step on any error.
+
+```yaml
+apiVersion: testworkflows.testkube.io/v1
+kind: TestWorkflow
+metadata:
+  name: overview--example-2
+spec:
+  steps:
+  - run:
+      image: 'curlimages/curl:7.78.0'
+      shell: |
+        curl https://testkube.io
+```        
+
+### Alternative Syntax
+
+Alternatively, you can use **shell** directly as an instruction and a default image will be used.
+
+![Using Shell](../img/using-shell.png)
+
+
+## Fetching the Git Repository
+
+### Fetching Data from Git
+
+To fetch data from the Git, use the git property of content. You may provide the address, revision (commit, branch or tag), along with i.e. token.
+
+To avoid fetching all files in mono-repository, you can use paths for sparse checkout.
+
+```yaml
+apiVersion: testworkflows.testkube.io/v1
+kind: TestWorkflow
+metadata:
+  name: overview--example-3
+spec:
+  content:
+    git:
+      uri: 'https://github.com/kubeshop/testkube'
+      revision: 'main'
+      paths:
+      - 'test/cypress/executor-tests/cypress-12'
+
+  steps:
+  - shell: 'tree /data/repo'
+```  
+
+### Mounting
+
+By default the repository is mounted to /data/repo directory. You can control it with mountPath property though.
+
+![Mounting](../img/mounting.png)
+
+## Fetching a Secured Git Repository
+
+### Usinga Token from A Secret
+
+You may provide the Git token and username either as plain-text, or via the tokenFrom and usernameFrom clauses that are the same as native Kubernetes env.*.valueFrom.
+
+```yaml
+apiVersion: testworkflows.testkube.io/v1
+kind: TestWorkflow
+metadata:
+  name: overview--example-4
+spec:
+  content:
+    git:
+      uri: 'https://github.com/kubeshop/testkube'
+      tokenFrom:
+        secretKeyRef:
+          name: 'private-repo-secret-token'
+          key: 'token'
+
+  steps:
+  - shell: 'tree /data/repo'
+```
+
+![Secret Token](../img/secret-token.png)
```

Comment:
The image here is not helpful I guess, as it's the same as for just loading the repo